mesa/experimental/cell_space/cell.py

```diff
@@ -80,7 +87,9 @@ def disconnect(self, other: Cell) -> None:
             other (Cell): other cell to remove from connections
 
         """
-        self._connections.remove(other)
+        keys_to_remove = [k for k, v in self.connections.items() if v == other]
+        for key in keys_to_remove:
+            del self.connections[key]
```

Comment:
Can't this be simplified. If the name is just the coordinate of the cell, can't you do something like

self.connections.pop(other.coordinate)

Since removal is likely to be rare, not something to be overly worried about.

Reply:
No, the name is not the absolute coordinate of other, but a relative coordinate of self, so unfortunately this doesn't work